test-network-function/lifecycle/suite.go

```diff
@@ -300,6 +300,19 @@ func shutdownTest(podNamespace, podName string) {
 	test.RunAndValidate()
 }
 
+func testNodeDrain(env *config.TestEnvironment, nodeName string) {
+	// We need to delete all Oc sessions because the drain operation is often deleting oauth-openshift pod
+	// This results in lost connectivity for oc sessions
+	env.ResetOc()
+	// drain node
+	drainNode(nodeName)
+	// Ensure the node is uncordoned before exiting the function
+	defer uncordonNode(nodeName)
```

Comment:
Add a "uncordonNode(node)" on line 383 to uncordon the node if the drain command fails

Reply:
Yes, it is in the drainNode function. If the drain command fails, the node will remain cordoned
func drainNode(node string) {
	context := common.GetContext()
	tester := dd.NewDeploymentsDrain(drainTimeout, node)
	test, err := tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
	gomega.Expect(err).To(gomega.BeNil())
	result, err := test.Run()
	if err != nil || result == tnf.ERROR {
                *uncordonNode(node)*
		log.Fatalf("Test skipped because of draining node failure - platform issue")
	}
}